src/Tasks.UnitTests/GetCompatiblePlatform_Tests.cs

```diff
@@ -226,5 +226,30 @@ public void PlatformIsChosenAsDefault(string referencedPlatforms, string referen
             task.AssignedProjectsWithPlatform[0].GetMetadata("NearestPlatform").ShouldBe(string.Empty);
             task.Log.HasLoggedErrors.ShouldBeFalse();
         }
+        
+        // When `Platform` is retrieved in "GetTargetFrameworks" and that platform matches what the task has decided the project should be built as
+        // through negotiation. build that project _without_ a global property for Platform.
+        [Fact]
+        public void ChosenPlatformMatchesDefault()
+        {
+            TaskItem projectReference = new TaskItem("foo.bar");
+            projectReference.SetMetadata("Platforms", "AnyCPU");
```

Comment:
```suggestion
            projectReference.SetMetadata("Platforms", "AnyCPU;x64");
```

just to make sure this includes a "multiple platforms" scenario where none match.